pkg/app/app.go

```diff
@@ -1296,7 +1298,30 @@ func (a *App) apply(r *Run, c ApplyConfigProvider) (bool, bool, []error) {
 		}
 	}
 
-	if releasesToBeDeleted == nil && releasesToBeUpdated == nil {
+	var releasesWithPreApply []state.ReleaseSpec
```

Comment:
@Sajfer I think `releasesWithPreApply` is useless. see:
https://github.com/helmfile/helmfile/blob/1da88b829ccede343489e37e8dc8f9edd26a4b8c/pkg/event/bus.go#L52

Reply:
Can you explain more? `releasesWithPreApply` is what makes sure that the hook is run when doing `apply` even if there is no change to the release.

Reply:
@Sajfer I known your means.
1. move those codes to a sigle func:
```golang
	if len(releasesWithPreApply) > 0 {
		msg := "Releases with preapply hooks: \n"
		if infoMsg != nil {
			msg = fmt.Sprintf("%s%s", *infoMsg, msg)
		}
		infoMsg = &msg
	}
	for _, release := range releasesWithPreApply {
		tmp := fmt.Sprintf("%s  %s (%s)", *infoMsg, release.Name, release.Chart)
		infoMsg = &tmp
	}
```
2. if preapply has error. What we shoud do in the next?

Reply:
I would assume we should do the same thing as for any other hook that fails?

Reply:
ok. I think so.